scripts/docker_push.sh

```diff
@@ -0,0 +1,19 @@
+#!/bin/bash
+
+# Dockerhub login
+[ ! -z "$DOCKER_PASSWORD" ] || [ ! -z "$DOCKER_USERNAME" ];
+  echo "DOCKER_USERNAME and DOCKER_PASSWORD must be set as environment variables!";
+  exit 1;
+echo "$DOCKER_PASSWORD" | docker login -u "$DOCKER_USERNAME" --password-stdin
```

Comment:
Assuming there's no better way to read in a password here, use a [here string](https://www.tldp.org/LDP/abs/html/x17837.html) for `stdin` instead of invoking `echo` in a piped sequence: 
`docker login -u "$DOCKER_USERNAME" --password-stdin <<< "$DOCKER_PASSWORD"`

Reply:
I was pretty much following the TravisCI docs here: https://docs.travis-ci.com/user/docker/#pushing-a-docker-image-to-a-registry. What are the benefits of using a `here string` versus traditional piping?

Reply:
There's a lot of questionable Bash guidelines on the internet.  Using echo isn't horrible, but using the here-string form creates a simple temp file, and avoids creating a subshell, which has possible implications if defining variables before a command invocation.  So it's more of a "best practice" rather than "need-to-have".